File: app/src/main/java/fr/jbme/raiderioapp/data/model/character/MythicPlusRanks.java
package fr.jbme.raiderioapp.data.model.character;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class MythicPlusRanks {

    @SerializedName("overall")
    @Expose
    public Ranking overall;
    @SerializedName("class")
    @Expose
    public Ranking _class;
    @SerializedName("faction_overall")
    @Expose
    public Ranking factionOverall;
    @SerializedName("faction_class")
    @Expose
    public Ranking factionClass;
    @SerializedName("healer")
    @Expose
    public Ranking healer;
    @SerializedName("class_healer")
    @Expose
    public Ranking classHealer;
    @SerializedName("faction_healer")
    @Expose
    public Ranking factionHealer;
    @SerializedName("faction_class_healer")
    @Expose
    public Ranking factionClassHealer;
    @SerializedName("dps")
    @Expose
    public Ranking dps;
    @SerializedName("class_dps")
    @Expose
    public Ranking classDps;
    @SerializedName("faction_dps")
    @Expose
    public Ranking factionDps;
    @SerializedName("faction_class_dps")
    @Expose
    public Ranking factionClassDps;
    @SerializedName("spec_262")
    @Expose
    public Ranking spec262;
    @SerializedName("faction_spec_262")
    @Expose
    public Ranking factionSpec262;
    @SerializedName("spec_263")
    @Expose
    public Ranking spec263;
    @SerializedName("faction_spec_263")
    @Expose
    public Ranking factionSpec263;
    @SerializedName("spec_264")
    @Expose
    public Ranking spec264;
    @SerializedName("faction_spec_264")
    @Expose
    public Ranking factionSpec264;

}

File: app/src/main/java/fr/jbme/raiderioapp/model/RIOCharacter/MythicPlusRanks.kt
package fr.jbme.raiderioapp.model.RIOCharacter

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MythicPlusRanks(
    @SerializedName("overall")
    @Expose
    var overall: Ranking? = null,

    @SerializedName("class")
    @Expose
    var _class: Ranking? = null,

    @SerializedName("faction_overall")
    @Expose
    var factionOverall: Ranking? = null,

    @SerializedName("faction_class")
    @Expose
    var factionClass: Ranking? = null,

    @SerializedName("healer")
    @Expose
    var healer: Ranking? = null,

    @SerializedName("class_healer")
    @Expose
    var classHealer: Ranking? = null,

    @SerializedName("faction_healer")
    @Expose
    var factionHealer: Ranking? = null,

    @SerializedName("faction_class_healer")
    @Expose
    var factionClassHealer: Ranking? = null,

    @SerializedName("dps")
    @Expose
    var dps: Ranking? = null,

    @SerializedName("class_dps")
    @Expose
    var classDps: Ranking? = null,

    @SerializedName("faction_dps")
    @Expose
    var factionDps: Ranking? = null,

    @SerializedName("faction_class_dps")
    @Expose
    var factionClassDps: Ranking? = null,

    @SerializedName("spec_262")
    @Expose
    var spec262: Ranking? = null,

    @SerializedName("faction_spec_262")
    @Expose
    var factionSpec262: Ranking? = null,

    @SerializedName("spec_263")
    @Expose
    var spec263: Ranking? = null,

    @SerializedName("faction_spec_263")
    @Expose
    var factionSpec263: Ranking? = null,

    @SerializedName("spec_264")
    @Expose
    var spec264: Ranking? = null,

    @SerializedName("faction_spec_264")
    @Expose
    var factionSpec264: Ranking? = null
)